src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+     * This is actually a compacted (K,V) pair, and the hashCode and equality is on K only.
+     */
+    @DefaultSerializer(KmerAndCount.Serializer.class)
+    private final static class KmerAndCount extends SVKmer {
+        private int count;
+
+        KmerAndCount(final SVKmer kmer ) {
+            super(kmer);
+            this.count = 0;
+        }
+
+        private KmerAndCount(final Kryo kryo, final Input input ) {
+            super(kryo, input);
+            count = input.readInt();
+        }
+
+        protected void serialize( final Kryo kryo, final Output output ) {
+            super.serialize(kryo, output);
+            output.writeInt(count);
+        }
+
+        @Override
+        public boolean equals( final Object obj ) {
+            return obj instanceof KmerAndCount && equals((KmerAndCount)obj);
+        }
+
+        public boolean equals( final KmerAndCount that ) {
+            return super.equals(that);
+        }
+
+        public int getCount() { return count; }
+        public void incrementCount() { count += 1; }
+
+        @Override
+        public String toString() { return super.toString(SVConstants.KMER_SIZE)+" "+count; }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<KmerAndCount> {
+            @Override
+            public void write( final Kryo kryo, final Output output, final KmerAndCount kmerAndCount) {
+                kmerAndCount.serialize(kryo, output);
+            }
+
+            @Override
+            public KmerAndCount read(final Kryo kryo, final Input input, final Class<KmerAndCount> klass ) {
+                return new KmerAndCount(kryo, input);
+            }
+        }
+    }
+
+    private final static class KmerCounter implements Iterable<KmerAndCount> {
+        private static final int KMERS_PER_PARTITION_GUESS = 20000000;
+        private static final int MIN_KMER_COUNT = 2;
+        private final HopscotchHashSet<KmerAndCount> kmerSet;
+
+        KmerCounter( final Iterator<GATKRead> readItr ) {
+            kmerSet = new HopscotchHashSet<>(KMERS_PER_PARTITION_GUESS);
+            while ( readItr.hasNext() ) {
+                final Iterator<SVKmer> kmers = new SVKmerizer(readItr.next().getBases(),SVConstants.KMER_SIZE);
+                while ( kmers.hasNext() ) {
+                    kmerSet.put(new KmerAndCount(kmers.next().canonical(SVConstants.KMER_SIZE))).incrementCount();
+                }
+            }
+            final Iterator<KmerAndCount> kmerItr = kmerSet.iterator();
+            while ( kmerItr.hasNext() ) {
+                if ( kmerItr.next().getCount() < MIN_KMER_COUNT ) kmerItr.remove();
+            }
+        }
+
+        @Override
+        public Iterator<KmerAndCount> iterator() { return kmerSet.iterator(); }
+    }
+
+    /**
+     * Eliminates dups, and removes over-represented kmers.
+     */
+    private static final class KmerCleaner {
```

Comment:
Since this operates on single partitions only, it'd be helpful to note that in the comments and maybe class name. For example, as I read it, MAX_INTERVALS should really be MAX_INTERVALS_WITHIN_PARTITION (which could be a big difference).